test/integration/framework/etcd.go

```diff
@@ -207,6 +207,14 @@ func EtcdMain(tests func() int) {
 		// Both names occurred in practice.
 		goleak.IgnoreTopFunction("k8s.io/kubernetes/vendor/gopkg.in/natefinch/lumberjack%2ev2.(*Logger).millRun"),
 		goleak.IgnoreTopFunction("gopkg.in/natefinch/lumberjack%2ev2.(*Logger).millRun"),
+		// If there is an error during connection shutdown, SPDY keeps a
+		// goroutine around for ten minutes, which gets reported as a leak
```

Comment:
hmm, does this not potentially build up in prod as well? there's no way to override this timeout etc?

Reply:
No, it should not build up in prod. 1) This only happens if there is an error in the SPDY shutdown. Practically, this flake happened about 2 times in 600 tests. A script to reproduce the error ran the `portforward` integration test 1000 times, without triggering the goroutine leak. 2) The goroutine is cleaned up after 10 minutes. The timeout is long enough to trigger the leak detector, but it eventually the goroutine gets cleaned up.

Reply:
... and 3) we're removing SPDY anyhow :-)

Reply:
well, not completely :)

Reply:
@seans3 @BenTheElder :) https://github.com/moby/spdystream/issues/92

Reply:
> well, not completely :)

We are _completely_ removing it soon, aren't we? I don't think the goroutine leak is new?

Reply:
![image](https://github.com/kubernetes/kubernetes/assets/6450081/866edcc0-498c-478c-80b8-0da0a4b4a7b9)

Reply:
> We are completely removing it soon, aren't we?

we're tunnelling portforward spdy over websockets, because spdy has traffic direction / backpressure characteristics websockets don't

Reply:
and there are still two legs missing apiserver -- kubelet and kubelet -- container runtime

Reply:
I see, portforward-over-spdy-over-websocket, what a world :-)